iothub/device/src/IotHubBaseClient.cs

```diff
@@ -401,14 +401,10 @@ public async Task CloseAsync(CancellationToken cancellationToken = default)
             await InnerHandler.CloseAsync(cancellationToken).ConfigureAwait(false);
         }
 
-        /// <summary>
-        /// Releases the unmanaged resources used by the client and optionally disposes of the managed resources.
-        /// </summary>
-        /// <remarks>
-        /// The method <see cref="CloseAsync(CancellationToken)"/> should be called before disposing.
-        /// </remarks>
-        public void Dispose()
+        /// <inheritdoc/>
+        public async ValueTask DisposeAsync()
         {
+            await CloseAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
What if we encounter an exception as a part of the CloseAsync() call. Should we wrap this with a try block and add the Dipose() call in the finally block?

Reply:
Good question. I'm not sure that's what I'd do, but it is an interesting idea.

Anyone else want to weigh in?

Reply:
The SB client does what you suggest: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs#L85

Reply:
My initial thought is yes, we should have try/catch. But then I looked second round, this is on DisposeAsync() when we try to CloseAsync(). If there is actual exception that we can handle it differently, we should have try/catch. But it seems if catching an exception, we will ignore it and move on, right?

Reply:
Yes, it would silently fail. That's why I didn't originally like the idea, but that is what the SB client does.

Reply:
Oh, I take that back. They catch it only to log it, but then rethrow.

Reply:
I like the idea of us catching the exception, logging it and throwing it back to the user. Seems fair to me that since "close" fail, the resources also didn't get disposed.